### ISSUE TITLE
Do not remove trailing whitespace in Markdown files

### ISSUE THREAD

**Issue by HybridDog** [geany/geany#1464]
To edit .md files, l currently use leafpad instead of geany because l don't want to circuitously change the two settings which enable automatic removal of trailing whitespaces just to edit some README.md a bit.
In Markdown, two spaces at the end of the line cause a line break.

l'd like to edit Markdown files with geany without dreading vanished line breaks.

**Comment by elextr** [comment]
Unselect Menu->edit->preferences->files->strip trailing spaces and tabs

**Comment by HybridDog** [comment]
> l don't want to circuitously change the two settings which enable automatic removal of trailing whitespaces just to edit some README.md a bit.

**Comment by elextr** [comment]
Well, you may be able to write a plugin to detect a markdown file and change the setting when its saved, but its unlikely to be something to be included in Geany itself as its something specific to the filetype.

(And I managed to not say "invisible whitespace markup commands, what a stupid useless idiotic idea" ... ooops :)

**Comment by Bfgeshka** [comment]
ping.

Currently diff files are ignored in trailing spaces deletion, can it just be added here?

Or expose this option for filetype configs.

**Comment by elextr** [comment]
> Or expose this option for filetype configs.

Yeah, that would be the best solution, then it can be disabled by any filetype that needs to, not just hard coded built-in ones.

If somebody makes a pull request to do this it probably would be accepted (don't forget the manual).

**Comment by laoshaw** [comment]
geany has been my favorite editor for source code and text for years, but its markdown editing leaves so much in desire, the two space trailing is part of it. Please make geany a first class markdown editor, all github and git repo are a mix of code and markdown these days, geany needs a much better markdown component, be it a plugin or core component.

**Comment by elextr** [comment]
@laoshaw feel free to provide a pull request.  

Geany is a totally volunteer project, people work on what they need or are interested in.  To improve markdown or anything else it somebody who needs the improvement or is interested in it has to provide the changes.

**Comment by codebrainz** [comment]
Most editors don't have "first class" Markdown support any more than Geany, they require plugins/packages/bundles, for example:

https://atom.io/packages/search?q=markdown
https://github.com/SublimeText-Markdown/MarkdownEditing
https://marketplace.visualstudio.com/search?term=markdown&target=VS&category=All%20categories&vsVersion=vs15&sortBy=Relevance
https://wiki.gnome.org/Apps/Gedit/MarkdownSupport
https://github.com/textmate/markdown.tmbundle
https://plugins.geany.org/markdown.html

The thing they all have in common is being external to the core editor and being _contributed by users_ who needed these features, which is the whole point of having a plugin system in the first place.
